Shared/Samples/Edit feature attachments/EditFeatureAttachmentsView.swift

```diff
@@ -0,0 +1,359 @@
+    @State private var isAttachmentSheetPresented = false
+    
+    /// The data model for the sample.
+    @StateObject private var model = Model()
+    
+    var body: some View {
+        MapViewReader { mapProxy in
+            MapView(map: model.map)
+                .callout(placement: $model.calloutPlacement
+                    .animation(model.calloutShouldOffset ? nil : .default.speed(2))
+                ) { _ in
+                    HStack {
+                        CalloutView(model: model)
+                            .padding(6)
+                        Button("", systemImage: "exclamationmark.circle") {
+                            isAttachmentSheetPresented = true
+                        }
+                        .sheet(isPresented: $isAttachmentSheetPresented) {
+                            AttachmentSheetView(model: model)
+                        }
+                        .padding(8)
+                    }
+                }
+                .onDrawStatusChanged { drawStatus in
+                    // Updates the state when the map's draw status changes.
+                    if drawStatus == .completed {
+                        isLoaded = true
+                    }
+                }
+                .onSingleTapGesture { tap, _ in
+                    self.tapPoint = tap
+                }
+                .task(id: tapPoint) {
+                    guard let point = tapPoint else { return }
+                    model.featureLayer.clearSelection()
+                    do {
+                        let result = try await mapProxy.identify(
+                            on: model.featureLayer,
+                            screenPoint: point,
+                            tolerance: 5
+                        )
+                        guard let features = result.geoElements as? [ArcGISFeature],
+                              let feature = features.first else {
+                            return
+                        }
+                        try await model.setSelectedFeature(for: feature)
+                        if let location = mapProxy.location(fromScreenPoint: point) {
+                            model.updateCalloutPlacement(to: location)
+                        }
+                    } catch {
+                        self.error = error
+                    }
+                }
+                .overlay(alignment: .center) {
+                    if !isLoaded {
+                        ProgressView("Loading…")
+                            .padding()
+                            .background(.ultraThinMaterial)
+                            .cornerRadius(10)
+                            .shadow(radius: 50)
+                    }
+                }
+        }
+        .errorAlert(presentingError: $error)
+    }
+}
+
+private extension EditFeatureAttachmentsView {
+    // MARK: - AttachmentSheetView
+    
+    struct AttachmentSheetView: View {
+        /// The error shown in the error alert.
```

Comment:
@yo1995 should these be broken out into their own files?

Reply:
You can make the call. I do think the data model can be in a separate file as it is more than 100 lines.